<!doctype html>



  


<html class="theme-next mist use-motion" lang="zh-Hans">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>






<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.1" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="RNA," />








  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.1.1" />






<meta name="description" content="思考这样一个问题：“RNAseq原始的Pair-end测序数据质控之后，部分Pair-end的read变成了Single-end的read，分开比对后得到了PE的BAM和SE的BAM，这个时候要不要合并这两个BAM文件？”关于这个问题，我们在知识星球上对此进行过讨论。现在我总结一下我们的观点。 思考问题的熊： 这个问题可能需要多几个角度考虑。   为什么质控之后双端的reads变成了单端？一种">
<meta name="keywords" content="RNA">
<meta property="og:type" content="article">
<meta property="og:title" content="RNA-seq原始数据质控后,是否要合并PE和SE的比对结果|《解螺旋技术交流圈》精华第1期">
<meta property="og:url" content="http://huangshujia.me/2018/05/02/2018-05-02-Merge-The-Alignment-Result-Of-PE-And-SE-For-RNAseq-Or-Not.html">
<meta property="og:site_name" content="解螺旋的矿工">
<meta property="og:description" content="思考这样一个问题：“RNAseq原始的Pair-end测序数据质控之后，部分Pair-end的read变成了Single-end的read，分开比对后得到了PE的BAM和SE的BAM，这个时候要不要合并这两个BAM文件？”关于这个问题，我们在知识星球上对此进行过讨论。现在我总结一下我们的观点。 思考问题的熊： 这个问题可能需要多几个角度考虑。   为什么质控之后双端的reads变成了单端？一种">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="http://blog-fungenomics-com.qiniudn.com/choice-colors-colours-5933.jpg">
<meta property="og:image" content="http://blog-fungenomics-com.qiniudn.com/Trimmomatic_for_RNAseq.jpg">
<meta property="og:image" content="http://upload-images.jianshu.io/upload_images/2248079-8612d9b5e057bb24.jpg?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240">
<meta property="og:image" content="http://blog-fungenomics-com.qiniudn.com/helixminer_wechat_qr.png">
<meta property="og:updated_time" content="2018-05-27T02:22:13.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="RNA-seq原始数据质控后,是否要合并PE和SE的比对结果|《解螺旋技术交流圈》精华第1期">
<meta name="twitter:description" content="思考这样一个问题：“RNAseq原始的Pair-end测序数据质控之后，部分Pair-end的read变成了Single-end的read，分开比对后得到了PE的BAM和SE的BAM，这个时候要不要合并这两个BAM文件？”关于这个问题，我们在知识星球上对此进行过讨论。现在我总结一下我们的观点。 思考问题的熊： 这个问题可能需要多几个角度考虑。   为什么质控之后双端的reads变成了单端？一种">
<meta name="twitter:image" content="http://blog-fungenomics-com.qiniudn.com/choice-colors-colours-5933.jpg">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Mist',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":0,"b2t":false,"scrollpercent":false},
    fancybox: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://huangshujia.me/2018/05/02/2018-05-02-Merge-The-Alignment-Result-Of-PE-And-SE-For-RNAseq-Or-Not.html"/>





  <title>RNA-seq原始数据质控后,是否要合并PE和SE的比对结果|《解螺旋技术交流圈》精华第1期 | 解螺旋的矿工</title>
</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  





  <script type="text/javascript">
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?9091dc3354962cd0705c6a0c94b8a931";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>











  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">解螺旋的矿工</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            关于
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">
    <link itemprop="mainEntityOfPage" href="http://huangshujia.me/2018/05/02/2018-05-02-Merge-The-Alignment-Result-Of-PE-And-SE-For-RNAseq-Or-Not.html">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Shujia Huang">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="/images/avatar.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="解螺旋的矿工">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">RNA-seq原始数据质控后,是否要合并PE和SE的比对结果|《解螺旋技术交流圈》精华第1期</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2018-05-02T00:00:00+02:00">
                2018-05-02
              </time>
            

            

            
          </span>

          
            <span class="post-category" >
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/生物信息/" itemprop="url" rel="index">
                    <span itemprop="name">生物信息</span>
                  </a>
                </span>

                
                
                  ， 
                
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/生物信息/基因组学/" itemprop="url" rel="index">
                    <span itemprop="name">基因组学</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
              <span class="post-comments-count">
                <span class="post-meta-divider">|</span>
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
                <a href="/2018/05/02/2018-05-02-Merge-The-Alignment-Result-Of-PE-And-SE-For-RNAseq-Or-Not.html#comments" itemprop="discussionUrl">
                  <span class="post-comments-count disqus-comment-count"
                        data-disqus-identifier="2018/05/02/2018-05-02-Merge-The-Alignment-Result-Of-PE-And-SE-For-RNAseq-Or-Not.html" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          
             <span id="/2018/05/02/2018-05-02-Merge-The-Alignment-Result-Of-PE-And-SE-For-RNAseq-Or-Not.html" class="leancloud_visitors" data-flag-title="RNA-seq原始数据质控后,是否要合并PE和SE的比对结果|《解螺旋技术交流圈》精华第1期">
               <span class="post-meta-divider">|</span>
               <span class="post-meta-item-icon">
                 <i class="fa fa-eye"></i>
               </span>
               
                 <span class="post-meta-item-text">阅读次数 </span>
               
                 <span class="leancloud-visitors-count"></span>
             </span>
          

          

          

          

        </div>
      </header>
    

    <div class="post-body" itemprop="articleBody">

      
      

      
        <script src="/assets/js/APlayer.min.js"> </script><p><img src="http://blog-fungenomics-com.qiniudn.com/choice-colors-colours-5933.jpg" alt=""></p>
<h2 id="思考这样一个问题：“RNAseq原始的Pair-end测序数据质控之后，部分Pair-end的read变成了Single-end的read，分开比对后得到了PE的BAM和SE的BAM，这个时候要不要合并这两个BAM文件？”"><a href="#思考这样一个问题：“RNAseq原始的Pair-end测序数据质控之后，部分Pair-end的read变成了Single-end的read，分开比对后得到了PE的BAM和SE的BAM，这个时候要不要合并这两个BAM文件？”" class="headerlink" title="思考这样一个问题：“RNAseq原始的Pair-end测序数据质控之后，部分Pair-end的read变成了Single-end的read，分开比对后得到了PE的BAM和SE的BAM，这个时候要不要合并这两个BAM文件？”"></a>思考这样一个问题：“RNAseq原始的Pair-end测序数据质控之后，部分Pair-end的read变成了Single-end的read，分开比对后得到了PE的BAM和SE的BAM，这个时候要不要合并这两个BAM文件？”</h2><p>关于这个问题，我们在知识星球上对此进行过讨论。现在我总结一下我们的观点。</p>
<h2 id="思考问题的熊："><a href="#思考问题的熊：" class="headerlink" title="思考问题的熊："></a>思考问题的熊：</h2><blockquote>
<p>这个问题可能需要多几个角度考虑。 </p>
<ol>
<li>为什么质控之后双端的reads变成了单端？一种可能是因为一端的read质控确实不合格，第二种情况（通常也是更常见的）是因为因为建库的原因，去接头之后两条reads overlap，从信息量来说就变成了单端，这个时候类似trimmomatic的软件就会默认把其中一条read当作无用read而扔掉。 </li>
<li>通常来说，因为第一种情况而过滤掉的reads是相对少的，因此扔掉并无大碍，但是第二种就出问题了。目前大多数软件都不支持直接输入PE和SE的fastq文件进行mapping，你就必须把它分开做，这个时候你要是把单端的敢扔掉其实就是扔掉了大量的信息。 </li>
<li>怎么解决这个问题呢？目前看来比较简单的方法是直接将trimmomatic的一个参数keepBothReads 改为true ，让因为第二种原因而扔掉的reads得以保留，然后直接用pair的fastq做mapping就可以了。 </li>
<li>这个参数的影响可以看图<br><img src="http://blog-fungenomics-com.qiniudn.com/Trimmomatic_for_RNAseq.jpg" alt=""></li>
</ol>
</blockquote>
<h2 id="解螺旋的矿工（星球中的星主：YellowTree-）："><a href="#解螺旋的矿工（星球中的星主：YellowTree-）：" class="headerlink" title="解螺旋的矿工（星球中的星主：YellowTree+）："></a>解螺旋的矿工（星球中的星主：YellowTree+）：</h2><blockquote>
<p>RNAseq的数据在质控之后，有时候甚至会有多达10%的read会从Pair-end变为Single-End！！！丢掉的话，损失很大，所以要考虑留下来。我的看法和 @思考问题的熊 有相似之处，不过在处理Single-End的read上有些不同。我觉得可以这样做： </p>
<ol>
<li>过滤后，Pair-end read和Single-End Read分开各自去完成比对，得到各自的比对结果（BAM格式）；</li>
<li>分别计算PE的比对结果和SE比对结果在各个转录本上的覆盖数，然后把它们相加起来； </li>
<li>再用想在常用的基因表达分析工具（如EdgeR、DESeq等）进行下游分析； </li>
</ol>
<p>这里在（2）中唯一要担心的是，有些SE Read覆盖的转录本也许不是最准确的那个（因为缺了另一半，你无法有效去判断），但我觉得这部分是少数，对结果不会有影响，这就是我认为可以分开比对，分别计算覆盖数再合并的原因。这样做的另一个好处是，不用太担心另一半低质量的Read误导了比对结果（不过可能这种情况也不会太多）——这也就是我和 @思考问题的熊 存在差异的地方，不过我觉得熊的看法的一个好处是，处理起来会更简单一些。</p>
</blockquote>
<h2 id="CJ"><a href="#CJ" class="headerlink" title="CJ"></a>CJ</h2><blockquote>
<p>@YellowTree+， 支持哈。另，对于Trimmomatic处理之后产生大量单端数据，那么必然是size-selection这些步骤出问题咯。如果硬是要保留，我同样支持星主的做法。 @思考问题的熊，在此情况下，我觉得也同样有必要考虑Trimmomatic的其他参数，Trimmomatic默认的剪切模式会产生 正反完全 overlap 或者更过的情况。一些比对软件，如bowtie（或者hisat）似乎是不支持的。</p>
</blockquote>
<h2 id="矿工注解："><a href="#矿工注解：" class="headerlink" title="矿工注解："></a>矿工注解：</h2><p>CJ在这里指的是ILLUMINACLIP参数的“keepBothReads”参数。这个参数很重要，它的作用是R1和R2在去除了接头序列之后，如果剩余的部分是完全反向互补的，其默认参数（false），会把整条与R1完全反向互补的 R2去除，当做重复去除掉，但在有些情况下，例如这里需要用到Paired reads的时候，就要将这个参数改为 true，否则会损失一部分Paired reads。</p>
<p>我举个例子，看一个 PE150 数据的测试，就知道 keepBothReads 参数的重要性了：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div></pre></td><td class="code"><pre><div class="line">$ java -jar trimmomatic-0.36.jar PE -phred33 F-2-test_R1.fastq.gz F-2-test_R2.fastq.gz -baseout F-2.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:51</div><div class="line"></div><div class="line">ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences</div><div class="line">Input Read Pairs: 2500 Both Surviving: 1633 (65.32%) Forward Only Surviving: 828 (33.12%) Reverse Only Surviving: 12 (0.48%) Dropped: 27 (1.08%)</div><div class="line">TrimmomaticPE: Completed successfully</div><div class="line"><span class="comment"># 使用 ILLUMINACLIP 默认的第六个参数 false，只有 65.32% paired reads 保留下来</span></div><div class="line"></div><div class="line">$ java -jar trimmomatic-0.36.jar PE -phred33 F-2-test_R1.fastq.gz F-2-test_R2.fastq.gz -baseout F-2.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:8:<span class="literal">true</span> LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:51</div><div class="line"></div><div class="line">ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences</div><div class="line">Input Read Pairs: 2500 Both Surviving: 2439 (97.56%) Forward Only Surviving: 22 (0.88%) Reverse Only Surviving: 16 (0.64%) Dropped: 23 (0.92%)</div><div class="line">TrimmomaticPE: Completed successfully</div><div class="line"><span class="comment"># 将 ILLUMINACLIP 第六个参数改为 true，其余所有参数均相同，结果有 97.56% paired reads 保留下来</span></div></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2248079-8612d9b5e057bb24.jpg?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="知识星球"></p>
<hr>
<p>推荐阅读</p>
<ul>
<li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxOTUxOTM0Nw==&amp;mid=207310226&amp;idx=1&amp;sn=2a49ba8a7548e73e77fed8717c7d247e&amp;scene=21#wechat_redirect" target="_blank" rel="external">【干货】这么说，FPKM和RPKM真的是错的咯？！——关于FPKM/RPKM的深度反思</a></p>
</li>
<li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxOTUxOTM0Nw==&amp;mid=2649798425&amp;idx=1&amp;sn=ae355ed362848578e5c853413f23dfd7&amp;chksm=83c1d505b4b65c13124c9acd210356c4364ec9f5498bbd16fa4475be29811213abb64ea9720f&amp;scene=21#wechat_redirect" target="_blank" rel="external">GATK4.0和全基因组数据分析实践（上）</a></p>
</li>
<li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxOTUxOTM0Nw==&amp;mid=2649798455&amp;idx=1&amp;sn=67a7407980a57ce138948eb46992b603&amp;chksm=83c1d52bb4b65c3dde31df94e9686654bf616166c7311b531213ebf0010f67a32ce827e677b1&amp;scene=21#wechat_redirect" target="_blank" rel="external">GATK4.0和全基因组数据分析实践（下）</a></p>
</li>
<li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxOTUxOTM0Nw==&amp;mid=2649798366&amp;idx=1&amp;sn=b545fcea7f82839fa87e9d9e472d1e72&amp;chksm=83c1d4c2b4b65dd4843250c307969ada96c4039f4f528c034620d25b78d8beba2f9cf924bb8a&amp;scene=21#wechat_redirect" target="_blank" rel="external">该如何自学入门生物信息学</a></p>
</li>
</ul>
<hr>
<p>我的微信公众号：解螺旋的矿工 欢迎关注更及时了解更多信息。</p>
<p><img src="http://blog-fungenomics-com.qiniudn.com/helixminer_wechat_qr.png" alt="解螺旋的矿工"></p>
<hr>
<p>这是知识星球：<strong>解螺旋技术交流圈</strong>，是一个我与读者朋友们的私人朋友圈，欢迎你的加入。我有9年前沿而完整的生物信息学、NGS领域的工作经历，在该领域发有多篇Nature级别的科学文章。</p>
<p>这是知识星球上 <strong>第一个真正与基因组学和生物信息学强相关的圈子</strong>。我旨在营造一个高质量的组学知识圈和人脉圈，通过提问、彼此分享、交流经验、心得等，彼此更好地学习生信知识，提升基因组数据分析和解读的能力。</p>
<p>在这里你可以结识到全国优秀的基因组学和生物信息学专家，同时可以分享你的经验、见解和思考，有问题也可以向我提问和圈里的星友们提问。</p>
<p>知识星球邀请链接：<a href="https://wx.zsxq.com/mweb/views/joingroup/join_group.html?group_id=518881585444&amp;secret=vcdvs4rdpst7stq4wcvqmlwvogc0ssbn&amp;user_id=28821152428221" target="_blank" rel="external">「解螺旋技术交流圈」</a></p>

      
    </div>

    <div>
      
        

      
    </div>

    <div>
      
        

      
    </div>

    <div>
      
        

      
    </div>

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/RNA/" rel="tag"># RNA</a>
          
        </div>
      

      
        
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2018/04/30/2018-04-30-What-Is-Phasing.html" rel="next" title="人类基因组的Phasing原理是什么？">
                <i class="fa fa-chevron-left"></i> 人类基因组的Phasing原理是什么？
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/2018/05/11/2018-05-11-Code-For-GATK4-Best-Practice.html" rel="prev" title="GATK4全基因组数据分析最佳实践,我以这篇文章为标志,终结当前WGS系列数据分析的流程主体问题 | 完全代码">
                GATK4全基因组数据分析最佳实践,我以这篇文章为标志,终结当前WGS系列数据分析的流程主体问题 | 完全代码 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="disqus_thread">
        <noscript>
          Please enable JavaScript to view the
          <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a>
        </noscript>
      </div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap" >
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="/images/avatar.jpg"
               alt="Shujia Huang" />
          <p class="site-author-name" itemprop="name">Shujia Huang</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives">
                <span class="site-state-item-count">46</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">17</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">52</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="https://github.com/ShujiaHuang" target="_blank" title="GitHub">
                  
                    <i class="fa fa-fw fa-github"></i>
                  
                  GitHub
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="https://www.weibo.com/huangshujia2011" target="_blank" title="Weibo">
                  
                    <i class="fa fa-fw fa-weibo"></i>
                  
                  Weibo
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="https://www.zhihu.com/people/yellowtree" target="_blank" title="Zhihu">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                  Zhihu
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#思考这样一个问题：“RNAseq原始的Pair-end测序数据质控之后，部分Pair-end的read变成了Single-end的read，分开比对后得到了PE的BAM和SE的BAM，这个时候要不要合并这两个BAM文件？”"><span class="nav-number">1.</span> <span class="nav-text">思考这样一个问题：“RNAseq原始的Pair-end测序数据质控之后，部分Pair-end的read变成了Single-end的read，分开比对后得到了PE的BAM和SE的BAM，这个时候要不要合并这两个BAM文件？”</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#思考问题的熊："><span class="nav-number">2.</span> <span class="nav-text">思考问题的熊：</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#解螺旋的矿工（星球中的星主：YellowTree-）："><span class="nav-number">3.</span> <span class="nav-text">解螺旋的矿工（星球中的星主：YellowTree+）：</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#CJ"><span class="nav-number">4.</span> <span class="nav-text">CJ</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#矿工注解："><span class="nav-number">5.</span> <span class="nav-text">矿工注解：</span></a></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2018</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Shujia Huang</span>
</div>


<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>


        

        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.1"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.1"></script>



  
  

  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.1"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.1"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.1"></script>



  


  

    
      <script id="dsq-count-scr" src="https://huangshujia.disqus.com/count.js" async></script>
    

    
      <script type="text/javascript">
        var disqus_config = function () {
          this.page.url = 'http://huangshujia.me/2018/05/02/2018-05-02-Merge-The-Alignment-Result-Of-PE-And-SE-For-RNAseq-Or-Not.html';
          this.page.identifier = '2018/05/02/2018-05-02-Merge-The-Alignment-Result-Of-PE-And-SE-For-RNAseq-Or-Not.html';
          this.page.title = 'RNA-seq原始数据质控后,是否要合并PE和SE的比对结果|《解螺旋技术交流圈》精华第1期';
        };
        var d = document, s = d.createElement('script');
        s.src = 'https://huangshujia.disqus.com/embed.js';
        s.setAttribute('data-timestamp', '' + +new Date());
        (d.head || d.body).appendChild(s);
      </script>
    

  













  





  

  
  <script src="https://cdn1.lncld.net/static/js/av-core-mini-0.6.1.js"></script>
  <script>AV.initialize("PG9NA27zW2WuxmCMvOMlxzua-gzGzoHsz", "VMwO1UNTnT8FgGg411rzjALa");</script>
  <script>
    function showTime(Counter) {
      var query = new AV.Query(Counter);
      var entries = [];
      var $visitors = $(".leancloud_visitors");

      $visitors.each(function () {
        entries.push( $(this).attr("id").trim() );
      });

      query.containedIn('url', entries);
      query.find()
        .done(function (results) {
          var COUNT_CONTAINER_REF = '.leancloud-visitors-count';

          if (results.length === 0) {
            $visitors.find(COUNT_CONTAINER_REF).text(0);
            return;
          }

          for (var i = 0; i < results.length; i++) {
            var item = results[i];
            var url = item.get('url');
            var time = item.get('time');
            var element = document.getElementById(url);

            $(element).find(COUNT_CONTAINER_REF).text(time);
          }
          for(var i = 0; i < entries.length; i++) {
            var url = entries[i];
            var element = document.getElementById(url);
            var countSpan = $(element).find(COUNT_CONTAINER_REF);
            if( countSpan.text() == '') {
              countSpan.text(0);
            }
          }
        })
        .fail(function (object, error) {
          console.log("Error: " + error.code + " " + error.message);
        });
    }

    function addCount(Counter) {
      var $visitors = $(".leancloud_visitors");
      var url = $visitors.attr('id').trim();
      var title = $visitors.attr('data-flag-title').trim();
      var query = new AV.Query(Counter);

      query.equalTo("url", url);
      query.find({
        success: function(results) {
          if (results.length > 0) {
            var counter = results[0];
            counter.fetchWhenSave(true);
            counter.increment("time");
            counter.save(null, {
              success: function(counter) {
                var $element = $(document.getElementById(url));
                $element.find('.leancloud-visitors-count').text(counter.get('time'));
              },
              error: function(counter, error) {
                console.log('Failed to save Visitor num, with error message: ' + error.message);
              }
            });
          } else {
            var newcounter = new Counter();
            /* Set ACL */
            var acl = new AV.ACL();
            acl.setPublicReadAccess(true);
            acl.setPublicWriteAccess(true);
            newcounter.setACL(acl);
            /* End Set ACL */
            newcounter.set("title", title);
            newcounter.set("url", url);
            newcounter.set("time", 1);
            newcounter.save(null, {
              success: function(newcounter) {
                var $element = $(document.getElementById(url));
                $element.find('.leancloud-visitors-count').text(newcounter.get('time'));
              },
              error: function(newcounter, error) {
                console.log('Failed to create');
              }
            });
          }
        },
        error: function(error) {
          console.log('Error:' + error.code + " " + error.message);
        }
      });
    }

    $(function() {
      var Counter = AV.Object.extend("Counter");
      if ($('.leancloud_visitors').length == 1) {
        addCount(Counter);
      } else if ($('.post-title-link').length > 1) {
        showTime(Counter);
      }
    });
  </script>



  

  
  
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        tex2jax: {
          inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
          processEscapes: true,
          skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
        }
      });
    </script>

    <script type="text/x-mathjax-config">
      MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for (i=0; i < all.length; i += 1) {
          all[i].SourceElement().parentNode.className += ' has-jax';
        }
      });
    </script>
    <script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  


  

  

</body>
</html>
