<!doctype html>



  


<html class="theme-next mist use-motion" lang="zh-Hans">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>






<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.1" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="GATK," />








  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.1.1" />






<meta name="description" content="变异的质控，是我们在得到变异数据之后，接下来最重要的一个步骤。通常我们都是使用GATK VQSR模块来完成这个事情，关于VQSR的基本原理我在这篇文章中有写，但暂时不算详细。下面是大家经常都会用到的VQSR基本命令（以GATK4为例）： 12345678910111213141516171819202122232425262728293031323334353637383940414243##">
<meta name="keywords" content="GATK">
<meta property="og:type" content="article">
<meta property="og:title" content="我应该如何正确设置GATK VQSR的模型训练参数 |《解螺旋技术交流圈》精华第4期">
<meta property="og:url" content="http://huangshujia.me/2018/07/29/2018-07-29-Which-training-sets-arguments-should-I-use-for-running-vqsr.html">
<meta property="og:site_name" content="解螺旋的矿工">
<meta property="og:description" content="变异的质控，是我们在得到变异数据之后，接下来最重要的一个步骤。通常我们都是使用GATK VQSR模块来完成这个事情，关于VQSR的基本原理我在这篇文章中有写，但暂时不算详细。下面是大家经常都会用到的VQSR基本命令（以GATK4为例）： 12345678910111213141516171819202122232425262728293031323334353637383940414243##">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="http://blog-fungenomics-com.qiniudn.com/bird-s-eye-view-boats-colors.jpg">
<meta property="og:image" content="http://blog-fungenomics-com.qiniudn.com/hapmap.jpg">
<meta property="og:image" content="http://blog-fungenomics-com.qiniudn.com/1kgp_nature.jpeg">
<meta property="og:image" content="http://blog-fungenomics-com.qiniudn.com/helixminer-big-red.png">
<meta property="og:updated_time" content="2018-08-11T15:35:24.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="我应该如何正确设置GATK VQSR的模型训练参数 |《解螺旋技术交流圈》精华第4期">
<meta name="twitter:description" content="变异的质控，是我们在得到变异数据之后，接下来最重要的一个步骤。通常我们都是使用GATK VQSR模块来完成这个事情，关于VQSR的基本原理我在这篇文章中有写，但暂时不算详细。下面是大家经常都会用到的VQSR基本命令（以GATK4为例）： 12345678910111213141516171819202122232425262728293031323334353637383940414243##">
<meta name="twitter:image" content="http://blog-fungenomics-com.qiniudn.com/bird-s-eye-view-boats-colors.jpg">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Mist',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":0,"b2t":false,"scrollpercent":false},
    fancybox: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://huangshujia.me/2018/07/29/2018-07-29-Which-training-sets-arguments-should-I-use-for-running-vqsr.html"/>





  <title>我应该如何正确设置GATK VQSR的模型训练参数 |《解螺旋技术交流圈》精华第4期 | 解螺旋的矿工</title>
</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  





  <script type="text/javascript">
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?9091dc3354962cd0705c6a0c94b8a931";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>











  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">解螺旋的矿工</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            关于
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">
    <link itemprop="mainEntityOfPage" href="http://huangshujia.me/2018/07/29/2018-07-29-Which-training-sets-arguments-should-I-use-for-running-vqsr.html">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Shujia Huang">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="/images/avatar.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="解螺旋的矿工">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">我应该如何正确设置GATK VQSR的模型训练参数 |《解螺旋技术交流圈》精华第4期</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2018-07-29T00:00:00+08:00">
                2018-07-29
              </time>
            

            

            
          </span>

          
            <span class="post-category" >
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/生物信息/" itemprop="url" rel="index">
                    <span itemprop="name">生物信息</span>
                  </a>
                </span>

                
                
                  ， 
                
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/生物信息/基因组学/" itemprop="url" rel="index">
                    <span itemprop="name">基因组学</span>
                  </a>
                </span>

                
                
                  ， 
                
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/生物信息/基因组学/《解螺旋技术交流圈》精华/" itemprop="url" rel="index">
                    <span itemprop="name">《解螺旋技术交流圈》精华</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
              <span class="post-comments-count">
                <span class="post-meta-divider">|</span>
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
                <a href="/2018/07/29/2018-07-29-Which-training-sets-arguments-should-I-use-for-running-vqsr.html#comments" itemprop="discussionUrl">
                  <span class="post-comments-count disqus-comment-count"
                        data-disqus-identifier="2018/07/29/2018-07-29-Which-training-sets-arguments-should-I-use-for-running-vqsr.html" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          
             <span id="/2018/07/29/2018-07-29-Which-training-sets-arguments-should-I-use-for-running-vqsr.html" class="leancloud_visitors" data-flag-title="我应该如何正确设置GATK VQSR的模型训练参数 |《解螺旋技术交流圈》精华第4期">
               <span class="post-meta-divider">|</span>
               <span class="post-meta-item-icon">
                 <i class="fa fa-eye"></i>
               </span>
               
                 <span class="post-meta-item-text">阅读次数 </span>
               
                 <span class="leancloud-visitors-count"></span>
             </span>
          

          

          

          

        </div>
      </header>
    

    <div class="post-body" itemprop="articleBody">

      
      

      
        <script src="/assets/js/APlayer.min.js"> </script><p><img src="http://blog-fungenomics-com.qiniudn.com/bird-s-eye-view-boats-colors.jpg" alt=""></p>
<p>变异的质控，是我们在得到变异数据之后，接下来最重要的一个步骤。通常我们都是使用GATK VQSR模块来完成这个事情，关于VQSR的基本原理我在<a href="https://mp.weixin.qq.com/s/HeIhMeA6GNboQQl1b79arg" target="_blank" rel="external">这篇文章</a>中有写，但暂时不算详细。下面是大家经常都会用到的VQSR基本命令（以GATK4为例）：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div><div class="line">30</div><div class="line">31</div><div class="line">32</div><div class="line">33</div><div class="line">34</div><div class="line">35</div><div class="line">36</div><div class="line">37</div><div class="line">38</div><div class="line">39</div><div class="line">40</div><div class="line">41</div><div class="line">42</div><div class="line">43</div></pre></td><td class="code"><pre><div class="line"><span class="comment">## 首先是SNP mode</span></div><div class="line">time <span class="variable">$gatk</span> VariantRecalibrator \</div><div class="line">   -R <span class="variable">$reference</span>/Homo_sapiens_assembly38.fasta \</div><div class="line">   -V <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.vcf.gz \</div><div class="line">   -resource:hapmap,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">true</span>,prior=15.0 <span class="variable">$GATK_bundle</span>/hapmap_3.3.hg38.vcf \</div><div class="line">   -resource:omini,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">false</span>,prior=12.0 <span class="variable">$GATK_bundle</span>/1000G_omni2.5.hg38.vcf \</div><div class="line">   -resource:1000G,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">false</span>,prior=10.0 <span class="variable">$GATK_bundle</span>/1000G_phase1.snps.high_confidence.hg38.vcf \</div><div class="line">   -resource:dbsnp,known=<span class="literal">true</span>,training=<span class="literal">false</span>,truth=<span class="literal">false</span>,prior=2.0 <span class="variable">$GATK_bundle</span>/dbsnp_146.hg38.vcf \</div><div class="line">   -an DP -an QD -an FS -an SOR -an ReadPosRankSum -an MQRankSum \</div><div class="line">   -mode SNP \</div><div class="line">   -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 95.0 -tranche 90.0 \</div><div class="line">   -rscriptFile <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.plots.R \</div><div class="line">   --tranches-file <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.tranches \</div><div class="line">   -O <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.recal &amp;&amp; \</div><div class="line">time <span class="variable">$gatk</span> ApplyVQSR \</div><div class="line">   -R <span class="variable">$reference</span>/Homo_sapiens_assembly38.fasta \</div><div class="line">   -V <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.vcf.gz \</div><div class="line">   --ts_filter_level 99.0 \</div><div class="line">   --tranches-file <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.tranches \</div><div class="line">   -recalFile <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.recal \</div><div class="line">   -mode SNP \</div><div class="line">   -O <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.VQSR.vcf.gz &amp;&amp; <span class="built_in">echo</span> <span class="string">"** SNPs VQSR done **"</span></div><div class="line"></div><div class="line"><span class="comment">## 然后是Indel mode</span></div><div class="line">time <span class="variable">$gatk</span> VariantRecalibrator \</div><div class="line">   -R <span class="variable">$reference</span>/Homo_sapiens_assembly38.fasta \</div><div class="line">   -input <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.VQSR.vcf.gz \</div><div class="line">   -resource:mills,known=<span class="literal">true</span>,training=<span class="literal">true</span>,truth=<span class="literal">true</span>,prior=12.0 <span class="variable">$GATK_bundle</span>/Mills_and_1000G_gold_standard.indels.hg38.vcf \</div><div class="line">   -resource:dbsnp,known=<span class="literal">true</span>,training=<span class="literal">false</span>,truth=<span class="literal">false</span>,prior=2.0 <span class="variable">$GATK_bundle</span>/dbsnp_146.hg38.vcf \</div><div class="line">   -an DP -an QD -an FS -an SOR -an ReadPosRankSum -an MQRankSum \</div><div class="line">   -mode INDEL \</div><div class="line">   --max-gaussians 6 \</div><div class="line">   -rscriptFile <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.indels.plots.R \</div><div class="line">   --tranches-file <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.indels.tranches \</div><div class="line">   -O <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.indels.recal &amp;&amp; \</div><div class="line">time <span class="variable">$gatk</span> ApplyVQSR \</div><div class="line">   -R <span class="variable">$reference</span>/Homo_sapiens_assembly38.fasta \</div><div class="line">   -input <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.VQSR.vcf.gz \</div><div class="line">   --ts_filter_level 99.0 \</div><div class="line">   --tranches-file <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.indels.tranches \</div><div class="line">   -recalFile <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.snps.indels.recal \</div><div class="line">   -mode INDEL \</div><div class="line">   -O <span class="variable">$outdir</span>/poplation/<span class="variable">$&#123;outname&#125;</span>.HC.VQSR.vcf.gz &amp;&amp; <span class="built_in">echo</span> <span class="string">"** SNPs and Indels VQSR (<span class="variable">$&#123;outname&#125;</span>.HC.VQSR.vcf.gz finish) done **"</span></div></pre></td></tr></table></figure>
<p>在WGS系列文章中大家也可以到类似的程序操作命令。<strong>但是大多数初学者可能并不完全理解GATK VQSR中训练集参数（-resource）的内在含义</strong>，我以前在文章里也缺少对此进行过深入讨论。前段时间在知识星球里有几位小伙伴反复问到了这个问题，我一一作了回答，最后进行了总结。虽然这种参数是GATK VQSR模块所特有，但如果能理解好这些参数以及它们背后包含的意义，<strong>应该有助于我们更好地理解基因组的变异质控，更恰当地使用GATK。</strong>另外，对于生信算法开发者来说，还可以从这样的策略中或多或少地得到一些启发。</p>
<p>所以，这篇文章里，就把我在知识星球中有关这一块的内容整理出来分享给大家。</p>
<p>我们知道，<strong>SNP和Indel的特征是不同的，评价它们的好坏需要使用不同的标准。</strong>因此在使用VQSR进行变异质控的时候，<strong>它们各自的评估模型就需要分开训练和计算</strong>——我想大家在使用GATK分析WGS数据的时候就应该知道了（GATK通过参数-mode SNP或者-mode INDEL来有目的地选择SNP或者Indel，不需要自己去把VCF的SNP和Indel分出来），它们用于训练的数据集也是不完全一样的，所以下面我也按照这两个方面分开总结。</p>
<h2 id="SNP"><a href="#SNP" class="headerlink" title="SNP"></a>SNP</h2><p>对于SNP来说，VQSR的训练集数据目前主要有四个：HapMap，Omni，1000G和dbSNP，参数一般如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line">-resource:hapmap,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">true</span>,prior=15.0 hapmap_3.3.hg38.vcf \</div><div class="line">-resource:omni,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">false</span>,prior=12.0 1000G_omni2.5.hg38.vcf \</div><div class="line">-resource:1000G,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">false</span>,prior=10.0 1000G_phase1.snps.high_confidence.hg38.vcf \</div><div class="line">-resource:dbsnp,known=<span class="literal">true</span>,training=<span class="literal">false</span>,truth=<span class="literal">false</span>,prior=2.0 dbsnp_146.hg38.vcf \</div><div class="line">-an DP -an QD -an FS -an SOR -an ReadPosRankSum -an MQRankSum \</div><div class="line">-mode SNP \</div></pre></td></tr></table></figure>
<p>这个参数的格式，从左到右，按照逗号（,）隔开，分别是：</p>
<p>训练集名字，这个名字是可以随便改动的，但是为了便于交流，一般还是默认按照数据集的名字来设置（如上面的例子）；</p>
<ul>
<li>known：该数据是否作为已知变异数据集，用于对变异数据的标注；</li>
<li>training：该数据是否作为模型训练的数据集，用于训练VQSR模型；</li>
<li>truth：该数据是否作为验证模型训练的真集数据，这个数据同时还是VQSR训练bad model时自动进行参数选择的重要数据；</li>
<li>prior：该数据集在VQSR模型训练中的权重，或者叫Prior likelihood（这里转化为Phred-scale，比如20代表的是0.99）。</li>
</ul>
<p>一共有4个。<strong>但实际上，VQSR训练集的数据可以用-resource参数继续往下添加，有多少就加多少，对于人类WGS/WES数据来说，目前用的主要还是上面的这4个，</strong>但它们其实还是有些旧了（用了8年左右了），现在大人群的数据越来越多，这些数据集估计不久之后也会有变化，比如有些研究也开始加入gnomAD的数据。另外，如果你发现对你来说，这些数据集都不合适，比如都是非人的物种，那么应该自己按照需要构造这些数据集，不一定同样需要四个，只要保证有专门用来训练（training）和验证模型的真集（truth）数据就可以。</p>
<p>在上面的例子里，大家可能也注意到了，对于不同的数据集，与之对应的参数也是不同的，比如hapmap和1000G就不一样，除了prior值不同之外，known、training和truth的参数也不一样。<strong>但这些差异的实质和所代表的意义什么呢？在使用的时候到底是基于什么原则设定了这些参数？</strong></p>
<p><strong>毫无疑问的是这些差异的实质一定源自于不同的数据集本身</strong>，所以在逐一解释之前，我想先跟大家一起搞明白HapMap，Omni，1000G和dbSNP这四个数据集分别是什么，它们是怎么来的，以及它们各自都有什么特点。<strong>只要能够弄清楚这几点，那么参数的设置也会不言自明。</strong></p>
<p><strong>第一个是HapMap</strong>，它来自国际人类单倍体型图计划，HapMap的名字也是源自于此。这个项目刚启动之时，只有270样本，其中有60个家系。项目一共有三期，到第三期HapMap3的时候这个数据已经扩增到1301个样本了，其中有部分样本和千人基因组项目有重叠。<strong>由于这个数据集包含了大量家系数据，并且有非常严格的质控和严密的实验验证，因此它的准确性是目前公认最高的。</strong></p>
<p><img src="http://blog-fungenomics-com.qiniudn.com/hapmap.jpg" alt=""></p>
<p>所以VQSR进行质控模型训练的时候，会将其作为一个很重要的训练集（training=true）。它的权重也会被设置得很高，比如在WGS数据分析中常常设为prior=15——这里的Prior是Prior likelihood的Phred-scale，我们如果把15转换为likelihood，那么就是0.96838。此外，由于它的高准确性，通常还将作为模型验证的一个真集数据（truth=true）。</p>
<p><strong>第二个是Omni，</strong>这个数据源自Illumina的Omni基因型芯片，大概2.5百万个位点（我在知识星球中说是1M，这里纠正一下，应该是2.5M），它的验证结果常常作为基因型的金标准。比如用Omni芯片对千人基因组数据进行了验证的结果：1000G_omni2.5.hg38.vcf（这里hg38是参考序列版本），它也是一个高可信的变异结果，我们在VQSR模型训练的时候，同样可以为其设置很高的权重，一般为Prior=12(likelihood为0.9369)。<strong>通常情况下也可以把它作为验证结果的真集数据，</strong>但我这里所举的例子有些保守，把它设置为truth=false了，大家不必效仿，<strong>假如你没太大的“洁癖”把它设置为true都是没问题的（这也是GATK最佳实践的一般做法）。</strong></p>
<p><strong>第三个是1000G，</strong>这个数据从名字看我们也很熟悉，它就是千人基因组计划（1000 genomes project）质控后的变异数据，目前也是第三期，一共包含了2504个人的数据。通常来说质控后，它包含的绝大部分都是真实的变异，但由于没办法做全面的实验验证，并不能排除含有少部分假阳的结果。所以模型训练时给的权重虽然比较高——prior=10(likelihood为0.9)，但是一般就不作为模型验证的真集数据了，即truth=false。</p>
<p>﻿﻿<img src="http://blog-fungenomics-com.qiniudn.com/1kgp_nature.jpeg" alt=""></p>
<p><strong>第四个是dbSNP。</strong>说到dbSNP，<strong>这是一个绝对不可以作为训练集位点的数据——太脏了</strong>，为什么这么说呢？因为，<strong>dbSNP收集的数据，实际都是研究者们发表了相关文章提交上来的变异，这些变异很多是没做过严格验证的，很多甚至还是假的，在没被反复验证之前，是不可信的。</strong>因此，不会把它们作为模型的训练集，更不会把它作为真集看待（training=false,truth=false），权重也一般设置得很低，比如这里是prior=2（差不多才0.37）。dbSNP的唯一作用就是用于标注我们的变异集中哪些是已经在其它研究中出现过的——即属于已经被发现过的（已知）变异，给这些已知的变异位点标上RS id。</p>
<h2 id="Indel"><a href="#Indel" class="headerlink" title="Indel"></a>Indel</h2><p>对于Indel来说，VQSR模型的训练参数只有两个：Mill和dbSNP。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div></pre></td><td class="code"><pre><div class="line">-resource:mills,known=<span class="literal">true</span>,training=<span class="literal">true</span>,truth=<span class="literal">true</span>,prior=12.0 Mills_and_1000G_gold_standard.indels.hg38.vcf \</div><div class="line">-resource:dbsnp,known=<span class="literal">true</span>,training=<span class="literal">false</span>,truth=<span class="literal">false</span>,prior=2.0 dbsnp_146.hg38.vcf \</div><div class="line">-an DP -an QD -an FS -an SOR -an ReadPosRankSum -an MQRankSum \</div><div class="line">-mode INDEL \</div></pre></td></tr></table></figure>
<p><strong>第一个是Mills，</strong>对于Indel来说能正在算得上真集的并不多，Mills_and_1000G_gold_standard.indels.hg38.vcf算是其中一个，并被专门做过验证。但其实Indel并不那么容易验证，很多时候也是挑一些比较容易验证的结果。但不管如何，这是目前最佳的一个！所以权重各方面也都设置的比较高，比如prior=12，并且truth=true。</p>
<p><strong>第二个还是dbSNP，</strong>这个就和上面SNP模式下的作用是一样的。不过假如这一步对Indel进行VQSR的VCF数据是顺着上面SNP VQSR后下来的话，那么这个dbSNP的参数可以省略，因为已知变异的标注已经在SNP model下做好了。我在<a href="http://mp.weixin.qq.com/s?__biz=MzAxOTUxOTM0Nw==&amp;mid=2649798570&amp;idx=1&amp;sn=9445ed66605c8c62595458fa6b561e1d&amp;chksm=83c1d5b6b4b65ca0e3ec97082f65dbdcb6948aaae7560353115a5aa2dbd6ed7f8e233f34c3f1&amp;scene=21#wechat_redirect" target="_blank" rel="external">WGS系列的这篇文章</a>里，给出的流程就是这种情况。</p>
<h2 id="一些需要额外注意的地方"><a href="#一些需要额外注意的地方" class="headerlink" title="一些需要额外注意的地方"></a>一些需要额外注意的地方</h2><p>GATK VQSR在执行的时候要基于全基因组的所有变异数据，而不能拆分不同的染色体分别取执行，否则会导致各个染色体所训练的质控模型不一致。另外，除了要区分SNP和Indel模式之外，GATK VQSR分为两个步骤进行：VariantRecalibrator 和 ApplyVQSR，两者缺一不可。VariantRecalibrator用来进行模型计算，获得数据的情况，ApplyVQSR则是根据我们设定的ts_filter_level参数，最终过滤得到数据，这个参数基于我们对模型真集数据的灵敏度和特异性来确定，一般会设置为99.0%（比如上面例子）。但事实上，这个比例，并不是说它是最合适的，设定为99或者99.5本身并没有什么理论证明，更多还是一种约定俗成，或者大部分情况下，看到设置为这个值，结果看起来是合理的，而且是在正常人样本数据得到的认识。所以，有时候还是需要具体问题具体分析，多看VQSR得到的tranche图，特别是对于非人物种，如果你觉得95.0%也很合适，那么也不一定非得是99%。</p>
<hr>
<p><strong>你还可以看</strong></p>
<ul>
<li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxOTUxOTM0Nw==&amp;mid=2649798506&amp;idx=1&amp;sn=773a7db5dfd002c53fead86625266af5&amp;chksm=83c1d576b4b65c6057f93a00d62ea41364ccdf53429b1820442f5855a104dde68f4325376722&amp;scene=21#wechat_redirect" target="_blank" rel="external">人类基因组的Phasing原理是什么？</a></p>
</li>
<li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxOTUxOTM0Nw==&amp;mid=2649798604&amp;idx=1&amp;sn=48a2e16c6899880fc161cd071fd76651&amp;chksm=83c1d5d0b4b65cc68d27f58c16240a547e7180648a3965afca0917c0f59e4ad42cf62fc4ec46&amp;scene=21#wechat_redirect" target="_blank" rel="external">样本量重要，还是测序深度重要? 生物信息工程师可以分为多少种类型? </a></p>
</li>
<li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxOTUxOTM0Nw==&amp;mid=2649798570&amp;idx=1&amp;sn=9445ed66605c8c62595458fa6b561e1d&amp;chksm=83c1d5b6b4b65ca0e3ec97082f65dbdcb6948aaae7560353115a5aa2dbd6ed7f8e233f34c3f1&amp;scene=21#wechat_redirect" target="_blank" rel="external">GATK4全基因组数据分析最佳实践 ，我以这篇文章为标志，终结当前WGS系列数据分析的流程主体问题</a></p>
</li>
<li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxOTUxOTM0Nw==&amp;mid=2649798366&amp;idx=1&amp;sn=b545fcea7f82839fa87e9d9e472d1e72&amp;chksm=83c1d4c2b4b65dd4843250c307969ada96c4039f4f528c034620d25b78d8beba2f9cf924bb8a&amp;scene=21#wechat_redirect" target="_blank" rel="external">该如何自学入门生物信息学</a></p>
</li>
</ul>
<hr>
<p>我的微信公众号：<strong>解螺旋的矿工</strong></p>
<p><img src="http://blog-fungenomics-com.qiniudn.com/helixminer-big-red.png" alt="解螺旋的矿工"></p>
<hr>
<p>这是知识星球：『<strong>解螺旋技术交流圈</strong>』，是一个我与读者朋友们的私人朋友圈。我有9年前沿而完整的生物信息学、NGS领域的工作经历，在该领域发有多篇Nature级别的科学文章，我也希望借助这个知识星球把自己的一些微薄经验分享给更多对组学感兴趣的伙伴们。</p>
<p>这是知识星球上 <strong>第一个真正与基因组学和生物信息学强相关的圈子</strong>。我希望能够借此营造一个高质量的组学知识圈和人脉圈，通过提问、彼此分享、交流经验、心得等，<strong>彼此更好地学习生信知识，提升基因组数据分析和解读的能力</strong>。</p>
<p>在这里你可以结识到全国优秀的基因组学和生物信息学专家，同时可以分享你的经验、见解和思考，有问题也可以向我提问和圈里的星友们提问。</p>
<p>知识星球邀请链接：<a href="https://wx.zsxq.com/mweb/views/joingroup/join_group.html?group_id=518881585444&amp;secret=vcdvs4rdpst7stq4wcvqmlwvogc0ssbn&amp;user_id=28821152428221" target="_blank" rel="external">「解螺旋技术交流圈」</a></p>

      
    </div>

    <div>
      
        

      
    </div>

    <div>
      
        

      
    </div>

    <div>
      
        

      
    </div>

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/GATK/" rel="tag"># GATK</a>
          
        </div>
      

      
        
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2018/07/22/2018-07-22-Introduction-the-detection-of-structure-variants.html" rel="next" title="一篇文章说清楚基因组结构性变异检测的方法">
                <i class="fa fa-chevron-left"></i> 一篇文章说清楚基因组结构性变异检测的方法
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/2018/08/11/2018-08-11-What-is-insert-sequence.html" rel="prev" title="一篇文章说清楚什么是“插入片段”？">
                一篇文章说清楚什么是“插入片段”？ <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="disqus_thread">
        <noscript>
          Please enable JavaScript to view the
          <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a>
        </noscript>
      </div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap" >
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="/images/avatar.jpg"
               alt="Shujia Huang" />
          <p class="site-author-name" itemprop="name">Shujia Huang</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives">
                <span class="site-state-item-count">51</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">19</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">53</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="https://github.com/ShujiaHuang" target="_blank" title="GitHub">
                  
                    <i class="fa fa-fw fa-github"></i>
                  
                  GitHub
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="https://www.weibo.com/huangshujia2011" target="_blank" title="Weibo">
                  
                    <i class="fa fa-fw fa-weibo"></i>
                  
                  Weibo
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="https://www.zhihu.com/people/yellowtree" target="_blank" title="Zhihu">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                  Zhihu
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#SNP"><span class="nav-number">1.</span> <span class="nav-text">SNP</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Indel"><span class="nav-number">2.</span> <span class="nav-text">Indel</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#一些需要额外注意的地方"><span class="nav-number">3.</span> <span class="nav-text">一些需要额外注意的地方</span></a></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2018</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Shujia Huang</span>
</div>


<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>


        

        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.1"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.1"></script>



  
  

  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.1"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.1"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.1"></script>



  


  

    
      <script id="dsq-count-scr" src="https://huangshujia.disqus.com/count.js" async></script>
    

    
      <script type="text/javascript">
        var disqus_config = function () {
          this.page.url = 'http://huangshujia.me/2018/07/29/2018-07-29-Which-training-sets-arguments-should-I-use-for-running-vqsr.html';
          this.page.identifier = '2018/07/29/2018-07-29-Which-training-sets-arguments-should-I-use-for-running-vqsr.html';
          this.page.title = '我应该如何正确设置GATK VQSR的模型训练参数 |《解螺旋技术交流圈》精华第4期';
        };
        var d = document, s = d.createElement('script');
        s.src = 'https://huangshujia.disqus.com/embed.js';
        s.setAttribute('data-timestamp', '' + +new Date());
        (d.head || d.body).appendChild(s);
      </script>
    

  













  





  

  
  <script src="https://cdn1.lncld.net/static/js/av-core-mini-0.6.1.js"></script>
  <script>AV.initialize("PG9NA27zW2WuxmCMvOMlxzua-gzGzoHsz", "VMwO1UNTnT8FgGg411rzjALa");</script>
  <script>
    function showTime(Counter) {
      var query = new AV.Query(Counter);
      var entries = [];
      var $visitors = $(".leancloud_visitors");

      $visitors.each(function () {
        entries.push( $(this).attr("id").trim() );
      });

      query.containedIn('url', entries);
      query.find()
        .done(function (results) {
          var COUNT_CONTAINER_REF = '.leancloud-visitors-count';

          if (results.length === 0) {
            $visitors.find(COUNT_CONTAINER_REF).text(0);
            return;
          }

          for (var i = 0; i < results.length; i++) {
            var item = results[i];
            var url = item.get('url');
            var time = item.get('time');
            var element = document.getElementById(url);

            $(element).find(COUNT_CONTAINER_REF).text(time);
          }
          for(var i = 0; i < entries.length; i++) {
            var url = entries[i];
            var element = document.getElementById(url);
            var countSpan = $(element).find(COUNT_CONTAINER_REF);
            if( countSpan.text() == '') {
              countSpan.text(0);
            }
          }
        })
        .fail(function (object, error) {
          console.log("Error: " + error.code + " " + error.message);
        });
    }

    function addCount(Counter) {
      var $visitors = $(".leancloud_visitors");
      var url = $visitors.attr('id').trim();
      var title = $visitors.attr('data-flag-title').trim();
      var query = new AV.Query(Counter);

      query.equalTo("url", url);
      query.find({
        success: function(results) {
          if (results.length > 0) {
            var counter = results[0];
            counter.fetchWhenSave(true);
            counter.increment("time");
            counter.save(null, {
              success: function(counter) {
                var $element = $(document.getElementById(url));
                $element.find('.leancloud-visitors-count').text(counter.get('time'));
              },
              error: function(counter, error) {
                console.log('Failed to save Visitor num, with error message: ' + error.message);
              }
            });
          } else {
            var newcounter = new Counter();
            /* Set ACL */
            var acl = new AV.ACL();
            acl.setPublicReadAccess(true);
            acl.setPublicWriteAccess(true);
            newcounter.setACL(acl);
            /* End Set ACL */
            newcounter.set("title", title);
            newcounter.set("url", url);
            newcounter.set("time", 1);
            newcounter.save(null, {
              success: function(newcounter) {
                var $element = $(document.getElementById(url));
                $element.find('.leancloud-visitors-count').text(newcounter.get('time'));
              },
              error: function(newcounter, error) {
                console.log('Failed to create');
              }
            });
          }
        },
        error: function(error) {
          console.log('Error:' + error.code + " " + error.message);
        }
      });
    }

    $(function() {
      var Counter = AV.Object.extend("Counter");
      if ($('.leancloud_visitors').length == 1) {
        addCount(Counter);
      } else if ($('.post-title-link').length > 1) {
        showTime(Counter);
      }
    });
  </script>



  

  
  
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        tex2jax: {
          inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
          processEscapes: true,
          skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
        }
      });
    </script>

    <script type="text/x-mathjax-config">
      MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for (i=0; i < all.length; i += 1) {
          all[i].SourceElement().parentNode.className += ' has-jax';
        }
      });
    </script>
    <script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  


  

  

</body>
</html>
